client/alloc_runner.go

```diff
@@ -76,9 +76,20 @@ type AllocRunner struct {
 type allocRunnerState struct {
 	Version                string
 	Alloc                  *structs.Allocation
+	AllocDir               *allocdir.AllocDir
 	AllocClientStatus      string
 	AllocClientDescription string
-	Context                *driver.ExecContext
+
+	// Context is deprecated and only used to migrate from older releases.
```

Comment:
We use the following format for these: `// COMPAT: Remove in 0.X.0: <reason for compatibility code and when it can be removed`